### PR TITLE
switch to using ubi as the standard github build with Docker

### PR DESCRIPTION
**Pull request by dev-mlb**

### CODE REVIEW

File: .github/workflows/maven-ci.yml
Comment (cfkoehler) [inline .github/workflows/maven-ci.yml:40]: Not sure if we care. But now the only step that checks for changed files from the formatter is the mac build.

**Comment by cfkoehler** [comment]
Not needed here. But believe this changes a few of the steps in DEVELOPING.md for using docker
https://github.com/NationalSecurityAgency/emissary/blob/30f75f32565b0a07f6ef10bd843c441da7026117/DEVELOPING.md?plain=1#L615
Basically no longer needing to run a maven build first since it is done in the image build process.

**Review by cfkoehler — APPROVED** [review]

File: .github/workflows/maven-ci.yml
Comment (dev-mlb) [reply .github/workflows/maven-ci.yml:40]: I originally had it as a step in the Dockerfile to check, but I don't want it to fail if you're doing local development and testing.

Comment (dev-mlb) [reply .github/workflows/maven-ci.yml:40]: Separate runner??

Comment (cfkoehler) [reply .github/workflows/maven-ci.yml:40]: Feels a little overkill. But runners are free. So sure. Unless you think it is too much.

Comment (dev-mlb) [reply .github/workflows/maven-ci.yml:40]: Feels a little overkill, but sorta nice to have it as a separate runner. We could probably add things to it in the future if we wanted to? @jpdahlke Thoughts?

Comment (ldhardy) [reply .github/workflows/maven-ci.yml:40]: We've had PRs just to correct a branch that got merged that produces formatting changes.  Personally, I wish the build would fail sooner before the branch gets merged or even the PR gets created. But as long as this will make sure the changes get addressed before merge than it's probably fine to have it separated out.

Comment (cfkoehler) [reply .github/workflows/maven-ci.yml:40]: @ldhardy 
Here in emissary those formatting changes have blocked the merge since the CI check fails. Don't think we have needed those follow prs recently here. 

But yeah good point.

**Review by cfkoehler — DISMISSED** [review]

**Comment by jpdahlke** [comment]
This seems like a good change to me. Once this is in we can update the settings to require the actions to complete for merge.

**Comment by jpdahlke** [comment]
in favor of #765

**Pull request closed without merging**
